### DISCUSSION TITLE
Revolting is great but I don't want to cause discord.

### DISCUSSION THREAD

**Discussion by BinaryGenius** [stoatchat/discussions#377] [General]
I saw this on flathub today and it has peaked my interest. A private chatting app that is eerily similar to discord, blimey! Hopefully I can speak for some of the other people in the privacy community when I ask this. How is this better than matrix or xmpp? (pros and cons) Like is there end 2 end encryption can it bridge with signal or discord or any other service like matrix can. Does is finally solve the metadata issue? If Revolt really is better will youtubers or privacy advocates like techlore of even snowden recommend this to normal people? My goal is to get answers to most of these questions if possible and thanks for reading.

**Comment by I2rys** [comment]
How is this better than Matrix or XMPP?
First, though Revolt focuses in privacy It does not really focus in security unlike Matrix & XMPP. It's more of a Discord and since It's new maybe in the feature It will become much better anyways let's get started.

Matrix is a really good app for security & security, If you like to make servers much secure than Revolt & Discord then I recommend it.
Pros:
E2EE
Open Source
Decentralized

Cons:
Not good for communities It's more like a friends(Personal) chatting app but It's much better than XMPP
Lacks control of servers unlike Revolt & Discord
Lot's of people said that the company is bloated
Metadata is not encrypted in default

If you like security & privacy a lot then I recommend XMPP.
Pros:
E2EE
Decentralized
Open Source
According to my research they said XMPP is much better when It comes to privacy & security than Matrix
Metadata is encrypted and all other traffic

Cons:
Not good for communities It's more like a friends(Personal) chatting app.
Lacks control of servers unlike Revolt, Discord & Matrix.

For the last one Revolt. If you like a much secure version of Discord then the best alternative will be is Revolt, It's also much better when It comes to protecting your privacy as It can be self host.
Pros:
Open Source
Much better when It comes to handling communities unlike Matrix & XMPP

Cons:
Not decentralized
No E2EE
Metadata is not encrypted
You can't directly delete your account, you need to send an email to support

Is there an end 2 end encryption?
No, as for now the developers are focusing on adding new features & fixing bugs, but you can still suggest and see what they think.

Can it bridge with signal or discord or any other service like matrix can?
For now It can't be used to bridge with Signal but it can be used to bridge in Discord & Telegram. See https://github.com/insertish/awesome-revolt

If Revolt really is better will youtubers or privacy advocates like techlore of even snowden recommend this to normal people?
Good question but first of all Revolt is still under development/It's still new so we can't just jump to conclusion as we also don't know what they think about Revolt or have they used Revolt or do they know what is Revolt?

This is based on my research If you think I'm wrong please correct me and hope this answer your questions. <3

**Reply by BinaryGenius** [reply]
### Preface
Well first of all thanks for answering in the pros and cons format it helps readability for anyone who comes across this discussion later down the line. And I hope this doesn't come across as offensive but there are some small grammar errors here and there like "feature" instead of "future" or "security & security" instead of "privacy & security". I can still read your comment just fine but revising it might improve readability for future viewers.
### Needs
XMPP is great for privacy and security... that's all I can really say. There isn't a good client to recommend to people and the whole experience is a mess so for me it's off the table. I'm using matrix now but what people recommend to use is element. I've gotten a discord theme working with it but the flathub element app is very buggy and the servers are nothing compared to what discord offers. This issue is fixed on the website I believe but I would like to use a desktop app. The whole experience on matrix has been very bizarre so I'm looking for something new preferably with...
- E2EE + Metadata
- A clean and robust interface
- Friendly userbase
### Response
In all honesty decentralization is like a nice stamp on ones application I've never used it but it's nice if a service can accomplish it personally I never bother to self host but I could if needed so that's nice. I use DMs extensively so E2EE not being implemented is kind of a deal breaker for me hopefully we can get that soon! The interface looks clean from what I've seen discord is a clean looking app and revolt seems to follow their design so that's a check. I know this can't be controlled but with a good application there can be good moderation which leads to healthier and friendlier communities. Also you said Matrix lacks control of servers then you said under cons for XMPP...
> Lacks control of servers unlike Revolt, Discord & Matrix.

I'm confused about that one does it lack control of not?

Bridging would be nice down the line but I don't think it needs to be prioritized for now E2EE is much more important. Account deletion inside the application would be fantastic any good privacy app should essentially have a kill switch for the account. You did mention they delete the account through an email request that's fine for the time being but I hope they add an account delete button like discord.
### Closing
Techlore did talk about Revolt actually https://discuss.techlore.tech/t/matrix-community-closure/200/7 he mentions that it 'didn't meet his standards for building a community'. Of course these things take time but people are starting to notice Revolt and it could replace discord given enough effort. I feel like Revolt is ever so close to being a solid Privacy alternative but **all it needs is a little pushhhh**. If E2EE hasn't been suggested then consider it suggested because I just did that. I think all messaging apps should strive to be like signal. Stupid simple and very privacy and secure. The sad reality is no one adopts an application that isn't stupid simple; _at least that's what I've found_. Your answer was short and to the point so yeah I'd say it answered some of my questions. I might test it out with a burner account or something like that to see what I think aswell.

**Reply by strypey** [reply]
@I2rys
> This is based on my research If you think I'm wrong please correct me and hope this answer your questions.

I appreciate this pros/cons breakdown and it's mostly a good overview. But it does get a few things wrong IMHO.  Eg this one about matrix:

> Not good for communities It's more like a friends(Personal) chatting app

The opposite is true. Perhaps  @I2rys is getting confused with XMPP on this one? Direct Chats in matrix are just generic chat rooms with only two members. This has created all sorts of usability problems with using matrix apps just for chatting with friends, and this is not generally what it's used for. That said, matrix devs are aware this is a problem, and the UX of 1:1 chats has improved quite a bit over the last few years, at least in Element. 1:1 voice/ video calls are now excellent, for example.

Matrix was intended as, among other things, a decentralized IRC replacement. So it's designed to handle huge public chats, with permanent logging, and federation of room metadata and chat logs across participating homeservers. It's handling of both public and private chat rooms has always been streets ahead of anything in the XMPP world, where group chats were an afterthought (MUC), and usually dependent on the originating server.  [Matrix Spaces](https://matrix.org/blog/2021/05/17/the-matrix-space-beta/) - the equivalent of "servers" in Discord - were in beta a year before that pros/cons breakdown was posted.

Or this one about XMPP:

> Metadata is encrypted and all other traffic

Only some XMPP clients support E2EE at all, and of those, only some support OMEMO, which is currently the best practice method. Even among those that do, it often has to be manually turned on by the person using it. There is no encryption of metadata AFAIK, although the XMPP community are aware of the problem and working on XEPs to solve it.

"Some messaging systems will expose less metadata. Some will ensure and enforce end to end encryption. Some will allow for federation, which inevitably increases the metadata involved. Others will sacrifice server-side message handling entirely, trading user-experience for reduced exposure. The system-specific issues raised will affect any messaging system to some degree - sometimes better, and sometimes worse."

https://xmpp.org/about/myths/

FYI I posted [a brief intro to various federation protocol, including matrix and XMPP, in a discussion thread here about ActivityPub](https://github.com/orgs/revoltchat/discussions/469#discussioncomment-5976729).

One other thing;

> Lot's of people said that the company is bloated

Element is a VC-funded company and its true they currently have a strong influence on the development direction of the matrix protocol. But there is now a Matrix Foundation to steward protocol development, and now that more established chat companies like [ProcessOne - developers of ejabberd](https://www.process-one.net/blog/matrix-protocol-added-to-ejabberd/) - and  [RocketChat](https://www.rocket.chat/press-releases/rocket-chat-leverages-matrix-protocol-for-decentralized-and-interoperable-communications) are adopting matrix, it's likely it will evolve into a fully independent standards organisation for the matrix protocol.

**Comment by sussycatgirl** [comment]
> How is this better than matrix or xmpp?

Depends on your needs. Revolt is an open source, privacy friendly alternative to Discord while matrix and xmpp are decentralized chat protocols (not sure what exactly xmpp is but I assume it's similar to matrix).

> Is there end 2 end encryption?

E2EE is planned for DMs and the Saved Notes feature.

> Can it bridge with signal or discord or any other service like matrix can?

It's not built in, but we offer a bot API that allows you to build your own bridge service. For example, my bot AutoMod offers a Discord-Revolt bridge.

> Does is finally solve the metadata issue?

Most likely not but I can't answer this until E2EE goes live.

**Reply by phoebehfharris** [reply]
> Depends on your needs. Revolt is an open source, privacy friendly alternative to Discord while matrix and xmpp are decentralized chat protocols (not sure what exactly xmpp is but I assume it's similar to matrix).

XMPP is a generic messaging protocol. Often used for federated DMs like Matrix, but it can also be used for video conferencing, social media, push notifications,  etc.

**Reply by BinaryGenius** [reply]
>Depends on your needs. Revolt is an open source, privacy friendly alternative to Discord while matrix and xmpp are decentralized chat protocols (not sure what exactly xmpp is but I assume it's similar to matrix).

XMPP is borderline unusable for me because of the lack of uniformity and standardization? Idk if that's the right word basically I want a client that is robust and clean and usable with XMPP I couldn't find one i would confidently recommend to people. Matrix is what I use currently but I might try out revolt with an anon account to see if I like the interface. From my experience the communities on Matrix have been sort of toxic hopefully people are nicer on Revolt. And hopefully there will be a delete account button in the future.
> E2EE is planned for DMs and the Saved Notes feature.

I hope that comes soon I'd join the platform if E2EE was implemented since I use DMs extensively.
> It's not built in, but we offer a bot API that allows you to build your own bridge service. For example, my bot AutoMod offers a Discord-Revolt bridge.

I don't host servers but that sounds good I assume it just sends the message from the user's username maybe with the "bot" handle appended.
> Most likely not but I can't answer this until E2EE goes live.

Sounds like a huge shame this is Matrix's huge flaw hopefully that can be dealt with.

And considering your a collaborator I assume you help with the code. is the team accepting any contributions? Maybe I can help, I would like to learn some coding and I've been looking for a good excuse to get some more experience. This application looks really promising and if I can speed up the development in any way it could help me leave matrix and help more people leave discord for something better that cares about their privacy.

**Reply by sussycatgirl** [reply]
> And hopefully there will be a delete account button in the future.

We actually have one now!

![Screenshot_20220617-091329](https://user-images.githubusercontent.com/26145882/174245667-0a81058a-c635-468c-a215-579e8cdc5bf1.png)

> Is the team accepting any contributions?

Of course, although I think insert said he wants to implement E2EE himself. Feel free to help out with other issues/features though!

**Reply by BinaryGenius** [reply]
Oh that's awesome and I'm liking the interface already a step up from element.
